plugins/CoreHome/vue/src/ReportingPage/ReportingPage.vue

```diff
@@ -130,7 +136,9 @@ export default defineComponent({
         this.hasNoVisits = false;
       }
 
-      this.renderPage(newValue.category as string, newValue.subcategory as string);
+      setTimeout(() => {
+        this.renderPage(newValue.category as string, newValue.subcategory as string);
+      }, 0);
```

Comment:
This timeout actually fixes a problem with the no data notification not being shown at all.
The method is triggered within a `watch` on `date` and `period` in the URL. But within `renderPage`, the request for raw data is triggered using the date and period from URL. But the values are updated after the executing of `watch`, so doing it directly uses the old values for the request. Other solution to fix this would be to pass the new values of date and period to `renderPage`

Reply:
I suspect the combination of these two snippets to be the problem:

https://github.com/matomo-org/matomo/blob/9a3ef94df61682e7d543f3f8b438c368f6eb71a6/plugins/CoreHome/vue/src/MatomoUrl/MatomoUrl.ts#L56-L59
https://github.com/matomo-org/matomo/blob/9a3ef94df61682e7d543f3f8b438c368f6eb71a6/plugins/CoreHome/vue/src/AjaxHelper/AjaxHelper.ts#L730-L732

The `watch()` is fired when setting the new `MatomoUrl` value, but the `AjaxHelper` still sends the previous date from `Matomo.currentDateString` with the request. Because the `Matomo.currentDateString` is updated in `updatePeriodParamsFromUrl`, too late to be used.

If my findings are correct, I would say passing the required parameters to `renderPage` (and then forward to the ajax call) feels like a more intuitive way to solve this :thinking: Using a zero delay `setTimeout` always feels like "it fixes a problem, but no one will know what and why, and hopefully it will not be just removed" :see_no_evil:

Reply:
Changed that.

Reply:
The fix made it broken again by accident :(

The period/date parameters (and probably also `segment`?) need to passed on to `showOnlyRawDataMessageIfRequired`. And from there to both API calls (`VisitsSummary.getVisits` and `Live.getMostRecentVisitsDateTime`).

Right now the check to make the calls uses the correct parameters, but the calls themselves reuse the wrong ones from the not yet updated `MatomoUrl` instance :see_no_evil: